chia/consensus/blockchain.py

```diff
@@ -698,12 +699,16 @@ async def validate_unfinished_block_header(
         if len(block.transactions_generator_ref_list) > self.constants.MAX_GENERATOR_REF_LIST_SIZE:
             return None, Err.TOO_MANY_GENERATOR_REFS
 
-        if (
-            self.try_block_record(block.prev_header_hash) is None
-            and block.prev_header_hash != self.constants.GENESIS_CHALLENGE
-        ):
+        prev_b = self.try_block_record(block.prev_header_hash)
+        if prev_b is None and block.prev_header_hash != self.constants.GENESIS_CHALLENGE:
             return None, Err.INVALID_PREV_BLOCK_HASH
 
+        prev_tx_height = prev_tx_block(self, prev_b)
+
+        # With hard fork 2 we ban transactions_generator_ref_list.
+        if prev_tx_height >= self.constants.HARD_FORK2_HEIGHT and block.transactions_generator_ref_list != []:
+            return None, Err.TOO_MANY_GENERATOR_REFS
```

Comment:
I'm a bit puzzled as to why this is checked in multiple places. Maybe one is for all blocks and another is just for "transaction blocks"?